go/internal/feast/onlinestore/cassandraonlinestore.go

```diff
@@ -0,0 +1,392 @@
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	"github.com/gocql/gocql"
+	"github.com/golang/protobuf/proto"
+	"github.com/rs/zerolog/log"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+	gocqltrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/gocql/gocql"
+)
+
+type CassandraOnlineStore struct {
+	project string
+
+	// Cluster configurations for Cassandra/ScyllaDB
+	clusterConfigs *gocql.ClusterConfig
+
+	// Session object that holds information about the connection to the cluster
+	session *gocql.Session
+
+	config *registry.RepoConfig
+}
+
+type CassandraConfig struct {
+	hosts                   []string
+	username                string
+	password                string
+	keyspace                string
+	protocolVersion         int
+	loadBalancingPolicy     gocql.HostSelectionPolicy
+	connectionTimeoutMillis int64
+	requestTimeoutMillis    int64
+}
+
+func extractCassandraConfig(onlineStoreConfig map[string]any) (*CassandraConfig, error) {
+	cassandraConfig := CassandraConfig{}
+
+	// parse hosts
+	cassandraHosts, ok := onlineStoreConfig["hosts"]
+	if !ok {
+		cassandraConfig.hosts = []string{"127.0.0.1"}
+		log.Warn().Msg("host not provided: Using 127.0.0.1 instead")
+	} else {
+		var rawCassandraHosts []any
+		if rawCassandraHosts, ok = cassandraHosts.([]any); !ok {
+			return nil, fmt.Errorf("didn't pass a list of hosts in the 'hosts' field")
+		}
+		var cassandraHostsStr = make([]string, len(rawCassandraHosts))
+		for i, rawHost := range rawCassandraHosts {
+			hostStr, ok := rawHost.(string)
+			if !ok {
+				return nil, fmt.Errorf("failed to convert a host to a string: %+v", rawHost)
+			}
+			cassandraHostsStr[i] = hostStr
+		}
+		cassandraConfig.hosts = cassandraHostsStr
+	}
+
+	// parse username
+	rawUsername, ok := onlineStoreConfig["username"]
+	if !ok {
+		cassandraConfig.username = ""
+		log.Warn().Msg("username not defined, will not be using authentication")
+	} else {
+		cassandraConfig.username, ok = rawUsername.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert username to string: %v", rawUsername)
+		}
+	}
+
+	// parse password
+	rawPassword, ok := onlineStoreConfig["password"]
+	if !ok {
+		cassandraConfig.password = ""
+		log.Warn().Msg("password not defined, will not be using authentication")
+	} else {
+		cassandraConfig.password, ok = rawPassword.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert password to string: %v", rawPassword)
+		}
+	}
+
+	// parse keyspace
+	rawKeyspace, ok := onlineStoreConfig["keyspace"]
+	if !ok {
+		cassandraConfig.keyspace = "feast_keyspace"
+		log.Warn().Msg("keyspace not defined: Using 'feast_keyspace' as keyspace instead")
+	} else {
+		cassandraConfig.keyspace, ok = rawKeyspace.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert keyspace to string: %v", rawKeyspace)
+		}
+	}
+
+	// parse protocolVersion
+	protocolVersion, ok := onlineStoreConfig["protocol_version"]
+	if !ok {
+		protocolVersion = 4.0
```

Comment:
Minor thing but I remember facing some connection issues with 3.0 or without. Have you come across any issues when using any other Protocol version? Might be better to keep protocolVersion to 4.0